typescript/api/services/NamedQueryService.ts

```diff
@@ -253,26 +253,27 @@ export module Services {
             let query = {};
             if (_.isUndefined(value)) {
               if (queryParam.whenUndefined == NamedQueryWhenUndefinedOptions.defaultValue) {
-                if(queryParam.format == NamedQueryFormatOptions.days) {
-                  let days = _.toInteger(queryParam.defaultValue);
-                  let nowDateAddOrSubtract = moment();
-                  if (days > 0) {
-                    //Going forward in time X number of days
-                    nowDateAddOrSubtract = nowDateAddOrSubtract.add(days, 'days');
-                  } else if(days < 0) {
-                    //This "additional" step makes the code self explanatory
-                    days = days * -1;
-                    //Going backwards in time X number of days
-                    nowDateAddOrSubtract = nowDateAddOrSubtract.subtract(days, 'days');
-                  }
-                  value = nowDateAddOrSubtract.toISOString();
-                } else if(queryParam.format == NamedQueryFormatOptions.ISODate) {
```

Comment:
Looks like the ISODate branch is missing?

Reply:
The unit tests will confirm this but line 256 covers the behaviour for ISODate, no further calculations are needed to be made for ISODate, it just needs to set its default value only for the "days" option do we need to translate it to an ISO date string.